core/templates/pages/collection-player-page/collection-navbar/collection-navbar.component.spec.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for collection navbar component.
+ */
+
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { TestBed, ComponentFixture, waitForAsync } from '@angular/core/testing';
+import { CollectionNavbarComponent } from './collection-navbar.component';
+import { ReadOnlyCollectionBackendApiService } from 'domain/collection/read-only-collection-backend-api.service';
+import { UrlService } from 'services/contextual/url.service';
+
+describe('Collection navbar component', () => {
+  let us: UrlService;
+  let rocbas: ReadOnlyCollectionBackendApiService;
+  let component: CollectionNavbarComponent;
+  let fixture: ComponentFixture<CollectionNavbarComponent>;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule
+      ],
+      declarations: [
+        CollectionNavbarComponent
+      ],
+      providers: [
+        UrlService,
+        ReadOnlyCollectionBackendApiService
+      ],
+      schemas: [
+        NO_ERRORS_SCHEMA
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CollectionNavbarComponent);
+    component = fixture.componentInstance;
+    us = TestBed.inject(UrlService);
+    us = (us as unknown) as jasmine.SpyObj<UrlService>;
+    rocbas = TestBed.inject(ReadOnlyCollectionBackendApiService);
+    rocbas = (rocbas as unknown) as
```

Comment:
Why is this needed? Can't you directly spy on it without this?

Reply:
removed them